testplan/common/config/base.py

```diff
@@ -43,7 +70,7 @@ def ConfigOption(key, default=ABSENT):
     option after we exhaust the entity's containers.
     """
 
-    optional = Optional(key, default=default)
+    optional = _Optional(key, default=default)
```

Comment:
Can we not pass default to schema library at all, and handle the default value later in our own code?

Reply:
The current solution handle the default values in `Config.__init__`, it's our own code. Or do you mean put the `default` in `ConfigOption` but not `_Optional`, however, how can we get all of the `ConfigOption`s (including all that from parents) after validating, that's more complicated. Another reason is that schema libraries might be updated later and it might be able to handle inherited classes of `Optional` just like `Hook` (I think it should be)

Reply:
Now add a `custom_default` attributes to `Optional` instance and make the original argument `default` always ABSENT.